src/audio/host.c

```diff
@@ -377,14 +376,16 @@ static uint32_t host_get_copy_bytes_normal(struct comp_dev *dev)
 	if (!copy_bytes) {
 		if (dev->direction == SOF_IPC_STREAM_PLAYBACK)
 			comp_info(dev, "no bytes to copy, %d free in buffer, %d available in DMA",
-				  audio_stream_get_free_bytes(&hd->local_buffer->stream),
+				  audio_stream_get_free_bytes(&buffer->stream),
 				  avail_bytes);
 		else
 			comp_info(dev, "no bytes to copy, %d avail in buffer, %d free in DMA",
-				  audio_stream_get_avail_bytes(&hd->local_buffer->stream),
+				  audio_stream_get_avail_bytes(&buffer->stream),
 				  free_bytes);
 	}
 
+	buffer = buffer_release(buffer);
```

Comment:
...or maybe `buffer_release()` shouldn't be `__must_check` or maybe even return `void`. Wouldn't it be cleaner to use a separate `handle` variable for an "acquired buffer" and keep `buffer` as a generic object reference. But if they're both of type `struct buffer *` this creates danger of mistakenly accessing the object via the `buffer` pointer. Or you could make a separate `struct buffer_handle` type for it, but then you'd have to change more... Thinking out loud